src/mlpack/methods/ann/activation_functions/tanh_exponential_function.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * @file methods/ann/activation_functions/tanh_exponential_function.hpp
+ * @author Mayank Raj
+ *
+ * Definition and implementation of the Tanh exponential  function.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef TANH_EXPONENTIAL_FUNCTION_HPP_INCLUDED
+#define TANH_EXPONENTIAL_FUNCTION_HPP_INCLUDED
```

Comment:
We should follow the style of the rest of the code base for the naming here:

https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/ann/activation_functions/hard_sigmoid_function.hpp#L12-L13